packages/blade/src/components/ProgressBar/ProgressBarFilled.web.tsx

```diff
@@ -16,23 +17,87 @@ const pulse = keyframes`
   }
 `;
 
-const BoxWithProgressBarFilledTransition = styled(Box)<
+const slideAndScale = keyframes`
+  0% {
+    left: -8%;
+    transform: scaleX(100%);
+  }
+  50% {
+    transform: scaleX(500%);
+  }
+  100% {
+    left: 103%;
+    transform: scaleX(100%);
+  }
+`;
+
+const pulseAnimationStyles = ({
+  duration,
+  easing,
+  delay,
+}: {
+  duration: string;
+  easing: string;
+  delay: string;
+}): FlattenSimpleInterpolation => css`
+  height: 100%;
+  width: 100%;
+  opacity: 0;
+  background-color: white;
+  animation: ${pulse} ${duration} ${easing} infinite;
+  animation-delay: ${delay};
+`;
+
+const BoxWithFilledIndeterminateTransition = styled(Box)<
+  Pick<ProgressBarFilledProps, 'fillMotionDuration'>
+>(({ theme, fillMotionDuration }) => {
+  const duration = castWebType(makeMotionTime(getIn(theme.motion, fillMotionDuration)));
+  const easing = 'linear'; // TODO: Add this in motion tokens
```

Comment:
We will add this to our motion tokens later (not in this PR) once design adds this too

Reply:
https://github.com/razorpay/blade/issues/924